src/m3ninx/index/segment/builder/builder.go

```diff
@@ -288,15 +291,15 @@ func (b *builder) Docs() []doc.Document {
 	return b.docs
 }
 
-func (b *builder) FieldsIterable() segment.FieldsIterable {
+func (b *builder) FieldsIterable() segment.FieldsPostingsListIterable {
 	return b
 }
 
 func (b *builder) TermsIterable() segment.TermsIterable {
 	return b
 }
 
-func (b *builder) Fields() (segment.FieldsIterator, error) {
+func (b *builder) FieldsPostingsList() (segment.FieldsPostingsListIterator, error) {
```

Comment:
nit: Since we're replacing `segment.FieldsIterator` altogether now, maybe we just rename this back to `FieldsIterator` and the `FiledsPostingsListIterable` back to `FieldsIterable`?

Reply:
Seems like a more symmetric API with `TermsIterable` and `TermsIterator`.